client/src/shared/components/Dropdown/index.tsx

```diff
@@ -8,38 +7,60 @@ export interface Option {
 }
 
 export interface DropdownProps {
-  label?: string;
+  label: string;
   options: Option[];
+  classNames?: string;
 }
 
 const Dropdown: React.FC<DropdownProps> = ({
   options,
-  label
+  label,
+  classNames
 }: DropdownProps) => {
   const [selectedOption, setSelectedOption] = useState<string | undefined>(
     undefined
   );
   const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleDocumentClick = (event: MouseEvent): void => {
+      if (
+        dropdownRef.current != null &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener('click', handleDocumentClick);
+
+    return () => {
+      document.removeEventListener('click', handleDocumentClick);
+    };
+  }, []);
```

Comment:
- can you move this to separate hooks. This could be usable in the future.
- create reusable directory for hooks.
`src/shared/hooks/useOutsideClick.tsx`

this [documentation](https://www.robinwieruch.de/react-hook-detect-click-outside-component/) is very useful. I believe it's has similar implementation.